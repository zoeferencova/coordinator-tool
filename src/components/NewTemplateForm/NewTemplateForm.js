import React from 'react';
import { Button, Textarea, Input } from '../Utils/Utils'

import styles from './NewTemplateForm.module.css';

export default class NewTemplateForm extends React.Component {
    
    
    render() {
        return (
            <main>
                <form onSubmit={e => this.handlePostTemplate(e)}>
                    <h3>New Template Form</h3>
                    <div>
                        <label htmlFor="template_name">Name: </label>
                        <Input type="text" class="block" id="template_name"></Input>
                    </div>
                    <div>
                        <label htmlFor="template_subject">Subject: </label>
                        <Input type="text" class="block" id="template_subject"></Input>
                    </div>
                    <div>
                        <label htmlFor="template_body">Body: </label>
                        <Textarea name="template_body" id="template_body" cols="100" rows="30"></Textarea>
                    </div>
                    <Button>Cancel</Button>
                    <Button type="submit">Save Template</Button>
                </form>
            </main> 
        )
    }
}