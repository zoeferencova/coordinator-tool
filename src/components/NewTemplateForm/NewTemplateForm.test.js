import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewTemplateForm from './NewTemplateForm'

describe(`NewTemplateForm component`, () => {
  it('renders a .NewTemplateForm by default', () => {
    const wrapper = shallow(<NewTemplateForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})