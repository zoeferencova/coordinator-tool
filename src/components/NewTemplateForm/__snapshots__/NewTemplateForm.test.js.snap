// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTemplateForm component renders a .NewTemplateForm by default 1`] = `
<main>
  <form
    onSubmit={[Function]}
  >
    <h3>
      New Template Form
    </h3>
    <div>
      <label
        htmlFor="template_name"
      >
        Name: 
      </label>
      <Input
        class="block"
        id="template_name"
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="template_subject"
      >
        Subject: 
      </label>
      <Input
        class="block"
        id="template_subject"
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="template_body"
      >
        Body: 
      </label>
      <Textarea
        cols="100"
        id="template_body"
        name="template_body"
        rows="30"
      />
    </div>
    <Button>
      Cancel
    </Button>
    <Button
      type="submit"
    >
      Save Template
    </Button>
  </form>
</main>
`;
